File: core/DuplicateZeros/DuplicateZeros.java
package com.dbc.code;

import java.util.ArrayDeque;
import java.util.Queue;

public class DuplicateZeros {
    public void duplicateZeros(int[] arr) {
        Queue<Integer> queue = new ArrayDeque<>();
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == 0) queue.offer(0);
            queue.offer(arr[i]);
            arr[i] = queue.poll();
        }
    }
}

File: core/DuplicateZeros/DuplicateZeros.py
from typing import List
from collections import deque


def duplicateZeros(arr: List[int]) -> None:
    queue, length = deque(), len(arr)

    for index in range(length):
        if arr[index] == 0:
            queue.append(0)
        queue.append(arr[index])
        arr[index] = queue.popleft()
